Mathlib/GroupTheory/GroupAction/MultipleTransitivity.lean

```diff
@@ -0,0 +1,313 @@
+then pretransitivity descends from `n ↪ α` to `n ↪ β`.
+
+The subtlety is that if it is not injective, this map does not induce
+an equivariant map from `n ↪ α` to `n ↪ β`. -/
+@[to_additive]
+theorem IsPretransitive.of_embedding {n : Type*}
+    (hf : Function.Surjective f) [IsPretransitive G (n ↪ α)] :
+    IsPretransitive H (n ↪ β) where
+  exists_smul_eq x y := by
+    let aux (x : n ↪ β) : (n ↪ α) :=
+      x.trans (Function.Embedding.ofSurjective (⇑f) hf)
+    have aux_apply (x : n ↪ β) (i : n) : f.toFun (aux x i) = x i := by
+      simp only [trans_apply, aux]
+      apply Function.surjInv_eq
+    obtain ⟨g, hg⟩ := exists_smul_eq (M := G) (aux x) (aux y)
+    use σ g
+    ext i
+    rw [DFunLike.ext_iff] at hg
+    rw [smul_apply]
+    simp only [← aux_apply, ← hg, smul_apply, MulActionHom.map_smul']
+
+@[to_additive]
+theorem IsPretransitive.of_embedding_congr {n : Type*}
+    (hσ : Function.Surjective σ) (hf : Function.Bijective f) :
+    IsPretransitive G (n ↪ α) ↔ IsPretransitive H (n ↪ β) :=
+  isPretransitive_congr hσ hf.mulActionHom_embedding_isBijective
+
+section Zero
+
+/-- Any action is 0-pretransitive. -/
+@[to_additive]
+theorem is_zero_pretransitive {n : Type*} [IsEmpty n] :
+    IsPretransitive G (n ↪ α) := inferInstance
+
+/-- Any action is 0-pretransitive. -/
+@[to_additive]
+theorem is_zero_pretransitive' :
+    IsMultiplyPretransitive G α 0 := inferInstance
+
+
+end Zero
+
+section One
+
+/-- The equivalence `one ↪ α` with `α`, for `Unique one`. -/
+def _root_.Function.Embedding.oneEmbeddingEquiv {one : Type*} [Unique one] : (one ↪ α) ≃ α where
+  toFun f := f default
+  invFun a := {
+    toFun := fun _ ↦ a
+    inj' x y h := by simp [Unique.uniq inferInstance] }
+  left_inv f := by ext; simp [Unique.uniq]
+  right_inv a := rfl
+
+/-- For `Unique one`, the equivariant map from `one ↪ α` to `α`. -/
+@[to_additive "For `Unique one`, the equivariant map from `one ↪ α` to `α`"]
+def _root_.MulActionHom.oneEmbeddingMap {one : Type*} [Unique one] :
+    (one ↪ α) →[G] α := {
+  oneEmbeddingEquiv with
+  map_smul' _ _ := rfl }
+
+@[to_additive]
+theorem _root_.MulActionHom.oneEmbeddingMap_bijective {one : Type*} [Unique one] :
+    Function.Bijective (oneEmbeddingMap (one := one) (G := G) (α := α)) :=
+  oneEmbeddingEquiv.bijective
+
+/-- An action is 1-pretransitive iff it is pretransitive. -/
+@[to_additive]
+theorem oneEmbedding_isPretransitive_iff {one : Type*} [Unique one] :
+    IsPretransitive G (one ↪ α) ↔ IsPretransitive G α :=
+  isPretransitive_congr Function.surjective_id oneEmbeddingMap_bijective
+
+/-- An action is 1-pretransitive iff it is pretransitive. -/
+@[to_additive]
+theorem is_one_pretransitive_iff :
+    IsMultiplyPretransitive G α 1 ↔ IsPretransitive G α :=
+  oneEmbedding_isPretransitive_iff
+
+end One
+
+section Two
+
+/-- Two distinct elements of `α` give an embedding `Fin 2 ↪ α`. -/
+def _root_.Function.Embedding.embFinTwo {a b: α} (h : a ≠ b) : Fin 2 ↪ α where
```

Comment:
This (and the two results below) should be somewhere else, maybe in the new file.